docs/fsharp/style-guide/formatting.md

```diff
@@ -0,0 +1,506 @@
+let methods2 =
+    System.AppDomain.CurrentDomain.GetAssemblies()
+    |> List.ofArray
+    |> List.map (fun assm -> assm.GetTypes())
+    |> Array.concat
+    |> List.ofArray
+    |> List.map (fun t -> t.GetMethods())
+    |> Array.concat
+
+// OK
+let methods2 = System.AppDomain.CurrentDomain.GetAssemblies()
+               |> List.ofArray
+               |> List.map (fun assm -> assm.GetTypes())
+               |> Array.concat
+               |> List.ofArray
+               |> List.map (fun t -> t.GetMethods())
+               |> Array.concat
+
+// Not OK
+let methods2 = System.AppDomain.CurrentDomain.GetAssemblies()
+            |> List.ofArray
+            |> List.map (fun assm -> assm.GetTypes())
+            |> Array.concat
+            |> List.ofArray
+            |> List.map (fun t -> t.GetMethods())
+            |> Array.concat
+```
+
+### Formatting modules
+
+Code in a local module must be indented relative to the module, but code in a top-level module does not have to be indented. Namespace elements do not have to be indented.
+
+```fsharp
+// A is a top-level module.
+module A
+
+let function1 a b = a - b * b
+```
+
+```fsharp
+// A1 and A2 are local modules.
+module A1 =
+    let function1 a b = a*a + b*b
+
+module A2 =
+    let function2 a b = a*a - b*b
+```
+
+### Formatting object expressions and interfaces
+
+Object expressions and interfaces should be aligned in the same way with `member` being indented after 4 spaces.
+
+```fsharp
+let comparer =
+    { new IComparer<string> with
+          member x.Compare(s1, s2) =
+              let rev (s : String) =
+                  new String (Array.rev (s.ToCharArray()))
+              let reversed = rev s1 i
+              reversed.CompareTo (rev s2) }
+```
+
+### Formatting whitespace in expressions
+
+Avoid extraneous whitespace in F# expressions.
+
+```fsharp
+// OK
+spam (ham.[1])
+
+// Not OK
+spam ( ham.[ 1 ] )
+```
+
+Named arguments should also not have space surrounding the `=`:
+
+```fsharp
+// OK
+let makeStreamReader x = new System.IO.StreamReader(path=x)
+
+// Not OK
+let makeStreamReader x = new System.IO.StreamReader(path = x)
+```
+
+## Formatting blank lines
+
+* Separate top-level function and class definitions with two blank lines.
+* Method definitions inside a class are separated by a single blank line.
+* Extra blank lines may be used (sparingly) to separate groups of related functions. Blank lines may be omitted between a bunch of related one-liners (e.g. a set of dummy implementations).
+* Use blank lines in functions, sparingly, to indicate logical sections.
+
+## Formatting comments
+
+Generally prefer multiple double-slash comments over ML-style block comments.
+
+```fsharp
+// Prefer this style of comments when you want
+// to express written ideas on multiple lines.
+
+(*
+    Generally avoid these kinds of comments.
+*)
+```
+
+Start inline comments with a new line, and captialize the first letter.
+
+```fsharp
+let f x = x + 1 // Increment by one.
```

Comment:
'start inline comments with a new line" ? Can we indicate if this is bad (and if so, what's the ideal) ?
IMO sometimes an EOL // comment is appropriate and an important counterpoint to comments level with the code, which I'd normally use to indicate aht's about to happen